backend/src/main/java/ai/verta/modeldb/experiment/FutureExperimentDAO.java

```diff
@@ -559,4 +560,16 @@ private void updateVersionNumber(Handle handle, String expId) {
         .bind("exp_id", expId)
         .execute();
   }
+
+  public InternalFuture<GetTags.Response> getTags(String expId) {
+    return getProjectIdByExperimentId(Collections.singletonList(expId))
+        .thenCompose(
+            projectIdFromExperimentMap ->
+                futureProjectDAO.checkProjectPermission(
+                    projectIdFromExperimentMap.get(expId), ModelDBServiceActions.READ),
+            executor)
+        .thenCompose(unused -> tagsHandler.getTags(expId), executor)
+        .thenApply(tags -> tags.stream().sorted().collect(Collectors.toList()), executor)
```

Comment:
`tagsHandler` handles this, no?

Reply:
Yes it should.
Jira Ticket: https://vertaai.atlassian.net/browse/VUMM-207